### ISSUE TITLE
How is the performance evaluated when there is no test dataset available?

### ISSUE THREAD

**Issue by ssoyaavv** [thunlp/Prompt-Transferability#15]
Hello, Thank you for your work.  I have some questions arising from reviewing the paper and its code. Thank you in advance for your response.

1. In cases where a test dataset is not provided (e.g., similar to SQuAD and NQ_Open), how was the performance measured?
2. When examining the code, it seems that the validation dataset is being called during testing. Is my understanding correct?
(Note :  I have reviewed version 1 of the code.)

Regards.

**Comment by yushengsu-thu** [comment]
@ssoyaavv 

Since some datasets did not include a separate test set, we replaced the test set with the validation set in our experiments. Importantly, we focused on exploring the transferability of prompts rather than comparing the performance with other baselines in this paper. As a result, whether using the test set or validation set became less crucial.

Hope the above response can help you.

**Comment by ssoyaavv** [comment]
Thanks @yushengsu-thu

If so, did you not perform validation?

If then, c an you share exact number of epochs you used for the best checkpoint? 
I am particularly interested in SQuAD and NQ_OPEN.

If you performed validation, can you share how you configured the Validation data set?
I would appreciate it if you could let me know if there is no way that fits the example below.
1. Validation dataset == Test dataset
2.  Split train dataset and use it as validation dataset
	2-1. How did you set the size ratio of train and valid?

**Comment by yushengsu-thu** [comment]
@ssoyaavv 

The validation dataset is equivalent to the test dataset in this paper. 

Instead of splitting it myself, I downloaded the dataset from the huggingface/datasets and utilized their provided training and validation sets.

To find the best checkpoint, you can refer to the "copy_prompt_to_taskpromptdir.py" script. Specifically, I used epoch == 23 for SQuAD on the T5 model, and epoch == 15 for NQ_OPEN on the T5 model.

**Comment by ssoyaavv** [comment]
Thanks @yushengsu-thu 

To ensure the experiment setting is complete,  I want to make sure: 

1. Evaluation Metrics (Appendix `A.2`)
Is it correct that we used only F1 score for QA-task and ROUGE-L for SUM-task? Are EM, ROUGE1, and ROUGE2 not used at all?

2. Is the Source prompt and Target prompt used in `Figure 5-(a) T5Small` the same as the settings below?

> |Source task|Target task|
> |------|---|
> |SQuAD|Multi-News|
> |NQ-Open|Multi-News|
> |Multi-News|SAMSum|
> SAMSum|Multi-News|

3. Convergence Speedup (Appendix `B.3`)
3-1. I wonder how to measure Convergence Speedup. No relevant code was found on the Git Hub. 
Could you please share how you calculated Convergence Speedup and Comparable-result Speedup of QA and SUM in `Table 1-T5 XXL`?
3-2. I want to know the standard of **_"time of TPT reaching comparable result to PT"_**. 
Is the epoch in "copy_prompt_to_taskpromptdir.py"?

**Comment by yushengsu-thu** [comment]
@ssoyaavv 

Q1: Is it correct that we used only F1 score for QA-task and ROUGE-L for SUM-task? Are EM, ROUGE1, and ROUGE2 not used at all?
A1: Yes, I only use 1 score for QA-task and ROUGE-L for SUM-task.

Q2: Is the Source prompt and Target prompt used in Figure 5-(a) T5Small the same as the settings below?
A2: Excuse me. I didn't understand your question well (I might miss some important information). The experiment of Figure 5-(a) transfer prompts across 17 tasks and observe the transferability. Could you please explain your question more clear again for me~~

Q3: 
Q3-1: Calculated Convergence Speedup and Comparable-result Speedup of QA and SUM in Table 1-T5 XXL? 
A3-1: We record the performance every 100 steps during prompt tuning. If the performance didn't not keep raising (+-0.5%) in the following 3*100 steps, we see it has already converged. 

Q3-2.  The standard of `time of TPT reaching comparable result to PT`.
Is the epoch in "copy_prompt_to_taskpromptdir.py"?
A3-2: Nope, the epoch in copy_prompt_to_taskpromptdir.py is the `PT convergence time`. It's sorry that I neither upload the `time of TPT reaching comparable result to PT` nor keep it. 😢 😢 😢 

Hope these response can help you in your following works. If I did not express well with my words, and you require more detail to conduct your work, it's my pleaseure have an on-line meeting to resolve your problems.  Just throw me a mail and let me know.
My mail is: yushengsu.thu@gmail.com

**Comment by ssoyaavv** [comment]
I apologize for any confusion caused by my previous question. 

My intention was to ask about the Source Task and Target Task when conducting the Table1 - T5_XXL experiment.
the Performance (TPTTASK) (%) was measured under the settings mentioned above?

> |Target task|Source task|
> |------|---|
> |SQuAD|NQ-Open|
> |NQ-Open|Multi-News|
> |Multi-News|SAMSum|
> SAMSum|SQuAD|

**Comment by yushengsu-thu** [comment]
@ssoyaavv . Sorry to reply late. Yes, that's right. If you encounter any further problem, please kindly let me know.
